pyuvdata/beamfits.py

```diff
@@ -50,276 +50,276 @@ def read_beamfits(self, filename, run_check=True, check_extra=True,
-            if axis1_units == 'deg':
-                self.axis1_array = np.deg2rad(self.axis1_array)
-            elif axis1_units != 'rad':
-                raise ValueError('Units of first axis array are not "deg" or "rad".')
-            axis2_units = primary_header.pop('CUNIT' + str(ax_nums['img_ax2']), 'deg')
-            if axis2_units == 'deg':
-                self.axis2_array = np.deg2rad(self.axis2_array)
-            elif axis2_units != 'rad':
-                raise ValueError('Units of second axis array are not "deg" or "rad".')
-
-        n_efield_dims = max([ax_nums[key] for key in ax_nums])
-
-        if self.beam_type == 'power':
-            # check for case where the data is complex (e.g. for xy beams)
-            if n_dimensions > ax_nums['complex'] - 1:
-                complex_arrs = np.split(data, 2, axis=0)
-                self.data_array = np.squeeze(complex_arrs[0] + 1j * complex_arrs[1], axis=0)
+            self.beam_type = primary_header.pop('BTYPE', None)
+            if self.beam_type is not None:
+                self.beam_type = self.beam_type.lower()
             else:
-                self.data_array = data
+                bunit = primary_header.pop('BUNIT', None)
+                if bunit is not None and bunit.lower().strip() == 'jy/beam':
+                    self.beam_type = 'power'
 
-            # Note: This axis is called STOKES by analogy with the equivalent uvfits axis
-            # However, this is confusing because it is NOT a true Stokes axis,
-            #   it is really the polarization axis.
-            if primary_header.pop('CTYPE' + str(ax_nums['feed_pol'])).lower().strip() == 'stokes':
-                self.Npols = primary_header.pop('NAXIS' + str(ax_nums['feed_pol']))
-
-            self.polarization_array = np.int32(uvutils.fits_gethduaxis(primary_hdu,
-                                                                       ax_nums['feed_pol']))
-            self.set_power()
-        elif self.beam_type == 'efield':
-            self.set_efield()
-            if n_dimensions < n_efield_dims:
-                raise ValueError('beam_type is efield and data dimensionality is too low')
-            complex_arrs = np.split(data, 2, axis=0)
-            self.data_array = np.squeeze(complex_arrs[0] + 1j * complex_arrs[1], axis=0)
-            if primary_header.pop('CTYPE' + str(ax_nums['feed_pol'])).lower().strip() == 'feedind':
-                self.Nfeeds = primary_header.pop('NAXIS' + str(ax_nums['feed_pol']))
-            feedlist = primary_header.pop('FEEDLIST', None)
-            if feedlist is not None:
-                self.feed_array = np.array(feedlist[1:-1].split(', '))
-        else:
-            raise ValueError('Unknown beam_type: {type}, beam_type should be '
-                             '"efield" or "power".'.format(type=self.beam_type))
+            if self.beam_type == 'intensity':
+                self.beam_type = 'power'
```

Comment:
This particular diff is kind of baffling.  Huge gaps in the HEAD suggesting large deletions. The bits which are lined up don't match.